### PR TITLE
Add covalent reference doc

### PR DESCRIPTION
**Pull request by DannyS03**
- adds guide for using covalent api with polygon
- references covalent content

cc. @harishraisinghani

### CODE REVIEW

File: docs/develop/data/covalent.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  - index
+  - indexing
+  - query
+image: https://matic.network/banners/matic-network-16x9.png 
+---
+
+## Introduction
+
+Polygon brings massive scale to Ethereum using an adapted version of Plasma 
+with PoS based side chains that provides a solution for faster and extremely
```
Comment (cerberushades) [inline docs/develop/data/covalent.md:22]:
```suggestion
with PoS based sidechains that provides a solution for faster and extremely 
```

File: docs/develop/data/covalent.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+|Polygon Mumbai Testnet chainId|`80001`|
+|Polygon Blockchain Explorer|https://polygonscan.com/|
+|Block time|~3 seconds|
+|Data refresh latency|~6 seconds or 2 Blocks|
+
+</TableWrap>
+
+:::tip Quickstart
+
+Check out **[this introduction video](https://www.youtube.com/watch?v=qhibXxKANWE)**
```
Comment (cerberushades) [inline docs/develop/data/covalent.md:44]:
```suggestion
Check out **[<ins>this introduction video</ins>](https://www.youtube.com/watch?v=qhibXxKANWE)** 
```

File: docs/develop/data/covalent.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+:::
+
+## Supported endpoints
+
+All [__Class A__](https://www.covalenthq.com/docs/api/#tag--Class-A) endpoints are supported for the Matic mainnet and the Mumbai testnet. You can query either network via the unified API by changing the `chainId`.
+
+:::info Endpoints
+
+A full list of all the requests you are able to do on the Polygon network using Covalent
+are available on the [Covalent API documentation](https://www.covalenthq.com/docs/api/).
```
Comment (cerberushades) [inline docs/develop/data/covalent.md:56]:
```suggestion
are available on the [<ins>Covalent API documentation</ins>](https://www.covalenthq.com/docs/api/).
```

File: docs/develop/data/covalent.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+## Appendix
+
+### Matic Gas token
+
+To interact with the Matic network, MATIC tokens are required to pay as gas fees. Covalent's 
+responses automatically returns `gas_*` fields in the MATIC units.
+
+### Token mapping
+
+Covalent maintains an on-chain real-time mapping of token addresses between Ethereum mainnet and the Matic chain. These addresses are used to reverse-lookup prices on Matic and also to return the right token logo urls.
```
Comment (cerberushades) [inline docs/develop/data/covalent.md:71]:
```suggestion
Covalent maintains an on-chain real-time mapping of token addresses between the Ethereum mainnet and the Matic chain. These addresses are used to reverse-lookup prices on Matic and also to return the right token logo URLs.
```

Comment (cerberushades) [reply docs/develop/data/covalent.md:71]: Are we keeping this "Matic chain"?

File: docs/develop/data/covalent.md [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+### Token prices
+
+For tokens that have a mapping back to Ethereum mainnet, Covalent is able to return the mapped prices.
+
+### Infrastructure Providers
+
+The following provide infrastructure for this blockchain network:
+* [Chainstack](../../service-providers/chainstack)
+* [Ankr](../../service-providers/ankr)
+* [GetBlock](../../service-providers/getblock)
```
Comment (cerberushades) [inline docs/develop/data/covalent.md:89]: @DannyS03 they're all broken.

**Review by cerberushades — CHANGES_REQUESTED** [review]

File: docs/develop/data/covalent.md (hunk 5 above)
Comment (DannyS03) [reply docs/develop/data/covalent.md]: removed links as they are redundant

**Review by cerberushades — APPROVED** [review]

**Pull request merged**